services/blockchain-connector/shared/sdk/decoder.js

```diff
@@ -114,9 +114,17 @@ const decodeEventPayload = (eventName, payload) => {
 	return payload;
 };
 
+const decodeEventPayload = async (encodedEvent, schema) => {
+	const decodedEvent = schema && encodedEvent !== ''
+		? await codec.decode(schema, Buffer.from(encodedEvent, 'hex')) : '';
```

Comment:
```suggestion
	const decodedEvent = await codec.decode(schema, Buffer.from(encodedEvent, 'hex'));
```

Reply:
The decode call fails if the values are not set. should still remove the checks?

Reply:
applied.